### PR TITLE
[feat](binlog) Add Support recover binlog

### PR DESCRIPTION
**Pull request by Vallishp**
support recover table, recover partition
related PR:  https://github.com/apache/doris/pull/44818
Currently the  recover flow, we will be do partial sync, 

 to can not gurantee the data present in the recycle bin of target.

Testcases run and passed.
1) DbSync
   --- recover table
   --- recover table as new table.

   -- recover partition
   -- recover partition as new partition name.
      
2) TableSync 
   -- recover partition
   -- recover partition as new partition name.

### CODE REVIEW

**Review by dataroaring — APPROVED** [review]
LGTM

File: pkg/ccr/record/recover_info.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	if err != nil {
+		return nil, xerror.Wrap(err, xerror.Normal, "unmarshal create table error")
+	}
+
+	if recoverInfo.TableId == 0 {
+		return nil, xerror.Errorf(xerror.Normal, "table id not found")
+	}
+
+	/* need check for db level not supported.
+	 */
```
Comment (w41ter) [inline pkg/ccr/record/recover_info.go]: Please add the above check, and also check if the table name and partition name exist.

File: pkg/ccr/job.go [hunk 2]
```diff
@@ -2574,6 +2574,48 @@ func (j *Job) handleDropRollupRecord(commitSeq int64, dropRollup *record.DropRol
...
+
+	return j.handleRecoverInfoRecord(binlog.GetCommitSeq(), recoverInfo)
+}
+
+func (j *Job) handleRecoverInfoRecord(commitSeq int64, recoverInfo *record.RecoverInfo) error {
+	if j.isBinlogCommitted(recoverInfo.TableId, commitSeq) {
+		return nil
+	}
+
+	if recoverInfo.PartitionName == "" {
```
Comment (w41ter) [inline pkg/ccr/job.go]: Consider adding a helper function `IsRecoverPartition() bool { return record.PartitionId == -1 }`, and change below line to:

```suggestion
	if recoverInfo.IsRecoverPartition() {
```

**Review by w41ter — CHANGES_REQUESTED** [review]
Rest LGTM

**Review by w41ter — APPROVED** [review]

**Pull request merged**